aquadoggo/src/network/peers/behaviour.rs

```diff
@@ -110,32 +108,39 @@ impl Behaviour {
         message: SyncMessage,
     ) {
         let peer = Peer::new(peer_id, connection_id);
-        trace!(
-            "Notify swarm of received sync message: {} {}",
-            peer.display(),
-            message.display()
-        );
-        self.events
-            .push_back(ToSwarm::GenerateEvent(Event::MessageReceived(
-                peer, message,
-            )));
+        self.push_event(ToSwarm::GenerateEvent(Event::MessageReceived(
+            peer, message,
+        )));
+    }
+
+    fn push_event(&mut self, event: ToSwarm<Event, HandlerFromBehaviour>) -> bool {
+        if self.running {
+            self.events.push_back(event);
+            return true;
+        }
+        false
+    }
+
+    /// Stop the peers behaviour from running, it won't handle any connection events or received messages.
+    pub fn stop(&mut self) {
+        self.running = false
+    }
+
+    /// Restart the peers behaviour.
+    pub fn restart(&mut self) {
+        self.running = true
     }
```

Comment:
Ah, you already have that thing implemented we talked about during our meeting today! 😆  (I thought that `stop` and `restart` are some native network behaviour methods). Are you concerned about the established connections or what was the problem again?

Reply:
The problem is that during the client-relay handshake phase (where identify info is exchanged and circuit relay established) the peers behaviour understands any connections as being new replication peers and so initiates sync sessions and sends begins sending messages. I wanted to avoid this getting mixed up in the initial node setup phase and so added these methods to basically pause replication until we want it to start (after initialisation).

Reply:
Okay, cool. Why is then the second dial required?

Reply:
That triggers both peers to start replicating with each other. It's not beautiful, but it works!

Reply:
Okay, I'm adding a TODO then :+1: